core/procedures/create_sequence.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procedures
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/doltgresql/core"
+	"github.com/dolthub/doltgresql/core/sequences"
+)
+
+// CreateSequenceName is the name of the stored procedure for creating sequences.
+const CreateSequenceName = "doltgres_create_sequence"
```

Comment:
Shifting of responsibilities. Your `OpaqueDDL` suggestion accomplishes the same thing, just that we have support for stored procedures now instead of having to build out `OpaqueDDL`.